content/updates/understanding-the-terraform-check-block-feature.md

```diff
@@ -0,0 +1,112 @@
+This is where `check` blocks come into play, stepping in to handle assertions when the actual provisioning isn't the primary focus. Given the necessity of configuring external tools and writing tests in Ruby or Go, native HCL `check` blocks can be seen as a practical alternative. They can be used before running tests in sandbox environments — whether ephemeral or persistent, hosted in the cloud, or run locally using mocking tools.
+In fact, the main practical testing scenario might involve validating the provisioning of the proposed changes. In certain instances, examining a single attribute of the final state could be sufficient.
+
+
+## How this can be set up?
+Adding a simple check is trivial. Let’s assume we have a root module that manages AWS EKS cluster configuration and contains the resource `aws_eks_cluster`:
+```hcl
+resource "aws_eks_cluster" "default" {
+ name     = "my-cluster"
+ role_arn = var.role_arn
+ vpc_config {
+   subnet_ids = var.subnet_ids
+ }
+}
+```
+
+To start using `check` block, ensure you’re using Terraform v1.5+. It’s recommended to use a ~> constraint to set bounds on versions of Terraform and providers a [root module](https://developer.hashicorp.com/terraform/language/modules#the-root-module) depends on:
+
+```hcl
+terraform {
+ required_version = "~> 1.5"
+
+ required_providers {
+   aws = {
+     source  = "hashicorp/aws"
+     version = "~> 4.0"
+   }
+ }
+}
+```
+
+Let’s define a check that validates that our cluster instance is healthy and available:
+```hcl
+check "aws_eks_cluster_default" {
+ assert {
+   condition     = aws_eks_cluster.default.status == "ACTIVE"
+   error_message = "EKS cluster ${aws_eks_cluster.default.id} status is ${aws_eks_cluster.default.status}"
+ }
+}
+```
+
+## Using a data source in the assertions - In progress
+In addition to simple assertions, Terraform offers the ability to reference a data source within `check` block assertions. It’s queried by Terraform at the end of each operation to evaluate the checks and obtain the most recent data from your environment.
+Terraform’s operation won’t be interrupted by failures in the scoped data block or any unsuccessful assertions. This unlocks continuous validation of your assumptions about the infrastructure rather than being confined to the point of initial provisioning.
+
+Also, this allows to utilize [`external` data source](https://registry.terraform.io/providers/hashicorp/external/latest/docs/data-sources/external) provided by Hashicorp. It green-lights integration of arbitrary external scripts into your Terraform configuration. This can be a Python script, a shell script, or any other program that can read JSON from standard input and write JSON to standard output. It's worth noting that the use of the external data source comes with a caveat: it can potentially make your configuration depend on the specific environment where it runs, as it might rely on specific external scripts being available.
```

Comment:
```suggestion
Also, this allows to utilize [`external` data source](https://registry.terraform.io/providers/hashicorp/external/latest/docs/data-sources/external) provided by Hashicorp. It green-lights integration of arbitrary external scripts into your Terraform configuration. This can be a Python script, a shell script, or any other program that can read JSON from standard input and write JSON to standard output. It's worth noting that the use of the external data source comes with a caveat: it can potentially make your configuration depend on the specific environment where it runs, as it might rely on specific external scripts and language runtimes being available.
```